### PR TITLE
Fix SIMD field issues

### LINKED ISSUE #9108: JIT/Performance/CodeQuality/SIMD/RayTracer failing under JITStress=2
See e.g. https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x64_checked_windows_nt_jitstress2/68/

### PR DESCRIPTION
**Pull request by CarolEidt**
First, when a GT_FIELD is transformed to a GT_IND, if its GTF_FLD_NULLCHECK flag is set, it must be cleared. Having this set leads to assertions or worse in value numbering. It is not entirely clear why this is only causing issues for SIMD fields.

Second, when a SIMD assignment is re-morphed, the dest may have already been changed to a GT_IND, in which case no further morphing is required (it is already a simple assignment).

Fix #9108

### CODE REVIEW

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib PTAL
@sivarv this may address #9108 as well.
@stephentoub I believe this addresses https://github.com/dotnet/corefx/issues/15309.
cc @mellinoe @danmosemsft

**Comment by CarolEidt** [comment]
@dotnet-bot test Windows_NT jitstress2

**Comment by stephentoub** [comment]
> @stephentoub I believe this addresses dotnet/corefx#15309.

Excellent, thanks, Carol.  Do you think it addresses https://github.com/dotnet/coreclr/issues/9109 as well?

**Comment by CarolEidt** [comment]
I suspect so. The first of the two issues is what was causing strange behavior in `ValueNumStore::VNForRefInAddr`

**Comment by sivarv** [comment]
I have verified that the fix to morph.cpp:fgMorphOneAsgBlockOp() fixes RayTracer issue https://github.com/dotnet/coreclr/issues/9108

File: src/jit/morph.cpp [hunk 1]
```diff
@@ -6475,6 +6475,11 @@ GenTreePtr Compiler::fgMorphField(GenTreePtr tree, MorphAddrContext* mac)
                     addr->gtIntCon.gtFieldSeq = fieldSeq;
 
                     tree->SetOper(GT_IND);
+                    // The GTF_FLD_NULLCHECK is the same bit as GTF_IND_ARR_LEN.
+                    // We must clear it when we transform the node.
+                    // TODO-Cleanup: It appears that the GTF_IND_ARR_LEN flag is never checked, and note
```
Comment (sivarv) [inline src/jit/morph.cpp]: > GTF_IND_ARR_LEN flag [](start = 56, length = 21)

I guess you wanted to mean GTF_FLD_NULLCHECK flag is never checked here.

(From source search I see fgValuenumberTree() is checking for GTF_IND_ARR_LEN flag.

Comment (CarolEidt) [inline src/jit/morph.cpp]: Yes, you are right. That's what I meant. I'll update the comment.

**Comment by sivarv** [comment]
:shipit:

**Comment by danmosemsft** [comment]
Is a test feasible to prevent regression? I'm not sure of your team's practices.

**Comment by CarolEidt** [comment]
@dotnet-bot test Windows_NT jitstress2

**Comment by CarolEidt** [comment]
@dotnet-bot test Windows_NT corefx_baseline

**Comment by CarolEidt** [comment]
RayTracer fails with `COMPlus_JitStress=2` and `COMPlus_EnableAVX=0`, so I'm looking at that now.

**Comment by CarolEidt** [comment]
I have verified that the JitStress2 failure is a different (albeit related) issue, so I propose to merge this and file an issue (for which I have a preliminary fix) for the new JitStress2 failure.

**Comment by CarolEidt** [comment]
@BruceForstall could you look at this, since we've chatted offline a bit about it?

**Comment by CarolEidt** [comment]
@danmosemsft I am looking into adding a test.

File: src/jit/morph.cpp [hunk 2]
```diff
@@ -6514,6 +6519,11 @@ GenTreePtr Compiler::fgMorphField(GenTreePtr tree, MorphAddrContext* mac)
         }
     }
     noway_assert(tree->gtOper == GT_IND);
+    // The GTF_FLD_NULLCHECK is the same bit as GTF_IND_ARR_LEN.
+    // We must clear it when we transform the node.
+    // TODO-Cleanup: It appears that the GTF_FLD_NULLCHECK flag is never checked, and note
+    // that the logic above does its own checking to determine whether a nullcheck is needed.
```
Comment (briansull) [inline src/jit/morph.cpp]: We should eliminate GTF_FLD_NULLCHECK 
That can happen in a future checkin

**Comment by BruceForstall** [comment]
LGTM

**Pull request merged**